silkworm/core/types/transaction.hpp

```diff
@@ -87,6 +87,8 @@ struct Transaction : public UnsignedTransaction {
 
     // sender recovered from the signature
     std::optional<evmc::address> from{std::nullopt};
+    // store calculated hash
+    mutable std::optional<evmc::bytes32> cached_hash;
```

Comment:
I suggest to initialise `cached_hash` with `std::nullopt` as done for `from` field. This keeps the type definition more coherent and also seems to be still preferable [at least in GCC](https://godbolt.org/z/ooMq1TY8x)

Reply:
ok